control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -132,21 +133,24 @@ func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullS
 		return nil, fmt.Errorf("invalid OpenShift release version format: %s", imageConfig.Config.Labels["io.openshift.release"])
 	}
 
-	majorMinor := fmt.Sprintf("%s.%s", versionParts[0], versionParts[1])
-
-	operators := []string{
-		"certified-operators",
-		"community-operators",
-		"redhat-marketplace",
-		"redhat-operators",
+	operators := map[string]string{
+		"certified-operators": fmt.Sprintf("registry.redhat.io/redhat/certified-operator-index:v%s.%s", versionParts[0], versionParts[1]),
+		"community-operators": fmt.Sprintf("registry.redhat.io/redhat/community-operator-index:v%s.%s", versionParts[0], versionParts[1]),
+		"redhat-marketplace":  fmt.Sprintf("registry.redhat.io/redhat/redhat-marketplace-index:v%s.%s", versionParts[0], versionParts[1]),
+		"redhat-operators":    fmt.Sprintf("registry.redhat.io/redhat/redhat-operator-index:v%s.%s", versionParts[0], versionParts[1]),
 	}
 
-	urls := make(map[string]string)
-	for _, operator := range operators {
-		urls[operator] = fmt.Sprintf("registry.redhat.io/redhat/%s-index:v%s", operator, majorMinor)
+	_, err = registryclient.GetListDigest(ctx, operators["certified-operators"], pullSecret)
+	if err != nil {
```

Comment:
Should we be more exact here on the error message? 

Only get the minor version -1 when the error is "no tag or digest specified". If the error was "failed to get repo setup" decrementing the minor version isn't going to fix that issue and the error should be reported back to the user for example.